app/uk/gov/hmrc/excisemovementcontrolsystemapi/repository/BoxIdRepository.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.excisemovementcontrolsystemapi.repository
+
+import cats.implicits.toFunctorOps
+import org.apache.pekko.Done
+import org.mongodb.scala.model.{Filters, IndexModel, IndexOptions, Indexes, ReplaceOptions}
+import play.api.{Configuration, Logging}
+import uk.gov.hmrc.excisemovementcontrolsystemapi.config.AppConfig
+import uk.gov.hmrc.excisemovementcontrolsystemapi.repository.model.BoxIdRecord
+import uk.gov.hmrc.excisemovementcontrolsystemapi.repository.BoxIdRepository.mongoIndexes
+import uk.gov.hmrc.excisemovementcontrolsystemapi.utils.DateTimeService
+import uk.gov.hmrc.mongo.MongoComponent
+import uk.gov.hmrc.mongo.play.json.PlayMongoRepository
+import uk.gov.hmrc.play.http.logging.Mdc
+
+import java.util.concurrent.TimeUnit
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.duration.Duration
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class BoxIdRepository @Inject()
+(mongo: MongoComponent,
+ configuration: Configuration,
+ timeService: DateTimeService)(implicit ec: ExecutionContext) extends PlayMongoRepository[BoxIdRecord](
+  collectionName = "boxidretrievals",
+  mongoComponent = mongo,
+  domainFormat = BoxIdRecord.format,
+  indexes = mongoIndexes(configuration.get[Duration]("mongodb.boxId.TTL")),
+  replaceIndexes = true
+) with Logging {
+
+  def getBoxIdRecord(ern: String): Future[Seq[BoxIdRecord]] = Mdc.preservingMdc {
```

Comment:
Could this be called getBoxIds and just return a list of boxIds for the given ERN? The calling API will probably not care about getting back the same ern and lastUpdated, it will probably just want a list of boxIds rather than having to map a list of BoxIdRecords to a list of boxIds?

Reply:
oh yeah i suppose so, i thought it would want the lastUpdated but I suppose not